### DISCUSSION TITLE
imports multi-line grammar

### DISCUSSION THREAD

**Discussion by chawlanitin** [tliron/go-puccini#79] [General]
@tliron I am trying to parse a package that has an import definition in line with the example specified in Tosca
![image](https://user-images.githubusercontent.com/39600246/124136685-9ceaca00-da4a-11eb-895b-95350dd2b0c0.png)

```
imports:
- nodes:
     file: nodes.yml
```

but on parsing the package using puccini, I am getting the following error:
```
@18,3 imports[0].nodes: unsupported field
@18,3 imports[0].url: missing required field
```

Could you please confirm is this a bug in Puccini or the grammar used in the package is not correct?

**Comment by tliron** [comment]
I'm quite sure that the correct syntax is this:
```
imports:
- path1/path2/some_defs.yaml
- file: path1/path2/file2.yaml
  repository: my_service_catalog
  ...
```

**Reply by tliron** [reply]
By the way, you did discover a bug I will fix -- the "missing required field" report uses the wrong field name. In this case it's "file" not "url".

**Comment by jpc4242** [comment]
@tliron. Yes, what you show is the correct syntax according to TOSCA "definition", but the TOSCA document has an "autoincompatible" example. Is the same as substitution mapping being an array or a map that forced puccini to have a dedicated quirk.
I guess we should have a new quirk here, ONAP CSARS are using the "wrong" format because they based their code on the example provided in TOSCA document.
BTW: The "wrong" example appears in tosca 1.1 and tosca 1.2. It was corrected in tosca 1.3

**Reply by tliron** [reply]
Is ONAP still locked in TOSCA 1.2?

**Reply by jpc4242** [reply]
It will probably depend on which ONAP version. I am experiencing it right now with a big telco that has a project involving ONAP "ElAlto" and the generated files use TOSCA 1.1
EDIT: I performed a search on onap cloned repos. Mostly 1.1 although some files already have 1.2 or 1.3 (and no TOSCA 2.x)

**Reply by tliron** [reply]
OK, thanks. I will convert this into an issue and implement a quirk.

**Comment by tliron** [comment]
This has been implemented in #79 as "--quirk imports.sequencedlist". I hope you can verify that it works as expected!
